### ISSUE TITLE
pubsub: more eager to fail when calling .Receive in 1.10.2

### ISSUE THREAD

**Issue by ssp** [googleapis/google-cloud-go#3986]
**Client**
PubSub

**Environment**
Alpine Docker on GKE

**Go Environment**
$ go version
1.16.3
$ go env

**Code**
```go
pubsubClient.Subscription("subscription-name").Receive(ctx, …)
```

**Expected behavior**
Up to version 1.10.1, the pubsub library seemed to be quite resilient against missing network connections when calling `.Receive`.

I would expect that resilience to remain present across a bugfix-update.

**Actual behavior**
Since version 1.10.2, calling `.Receive` on a `Subscription` returns an error when having no network access.

The retrieval of the configuration introduced [here](https://github.com/googleapis/google-cloud-go/commit/1fcc78ac6ecb461c3bbede9667436614c9df1535#diff-820c00bfb025d27cd84a23cb38671b023d41f92431bea486d7d66a3c6eb9ccebR779) seems to trigger a remote call, returning an error if that fails.

**Additional context**

Waiting for a few seconds before calling `.Receive` worked around the issue for me (with the root cause probably being  a delay introduced to the network connections being available by the istio configuration), so this change in behaviour will not be apparent when running locally.

As this seems to be a significant change in the behaviour of the library that I didn’t expect in a bugfix release, I thought you may want to be aware of it in case it wasn’t intended.

**Comment by hongalex** [comment]
Thanks for reporting. Yeah, resiliency to network issues isn't something we guarantee in our libraries. With that said, this behavior should be intact again in v1.10.3 (specifically [this change](https://github.com/googleapis/google-cloud-go/pull/3985)), as I moved the config check further down so it's only used when we detect ordering keys in the messages received. Also, errors from that call are transparent now and won't be returned by the `Receive` call. Closing for now, but let me know if any of that needs clarification.
